pkg/datasource/datasource.go

```diff
@@ -33,6 +33,16 @@ const (
 	TypeMetrics
 )
 
+type dsError string
+
+func (err dsError) Error() string {
+	return string(err)
+}
+
+const (
+	ErrDiscard = dsError("discarded")
+)
+
```

Comment:
It does work, however it makes that a mutable variable. This way it's const. I've read a couple of articles around this, but I'm also torn because it's still heavily used in the standard library (maybe just because of backward compatibility - idk).

Reply:
Ok, I was surprised it was done in that way by the standard library, however it seems it's because of compatibility reasons https://github.com/golang/go/issues/63602#issuecomment-1767393825. 

For future PR: We could define 

```
type dsError string

func (err dsError) Error() string {
	return string(err)
}
```

in a helper file and use it from other places, right?

Reply:
Yeah, that type would ofc need to be exported, then. But I think it's small enough to be kept inside the pkg it is being used from.